pkg/acsengine/engine.go

```diff
@@ -1791,6 +1797,9 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 			}
 			return s
 		},
+		"subtract": func(a, b int) int {
```

Comment:
Would it be possible to use the [Azure resource manager subtract function](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-numeric#sub) instead here?

Reply:
Hi @CecileRobertMichon this `subtract` function in engine.go is called by the go template in kubernetesmasterresources.t https://github.com/ritazh/acs-engine/blob/ea2e167f29eba114148d3fa72b9df51d6ed5e193/parts/k8s/kubernetesmasterresources.t#L753-L786 to generate all the arm resources under `dependsOn` in the arm template. I'm not aware of any way to mix results from the arm function`sub` and the go template used to generate arm resources. Please let me know if this is done incorrectly or maybe an example of where the two are mixed. Thanks!!!

Reply:
You're right, I thought this was used as an ARM function... In this case, the [go template package](https://golang.org/pkg/text/template/) doesn't have a subtract function AFAIK. The [sprig package](https://godoc.org/github.com/Masterminds/sprig) does but I don't think it's worth blocking this PR for it, we could add an import for the sprig package and use it instead of defining our own function as a follow up.